mod-clans/src/main/java/gg/solarmc/loader/clans/ClanManager.java

```diff
@@ -56,45 +53,59 @@
         return Optional.ofNullable(clans.getIfPresent(id));
     }
 
+
     /**
-     * Gets a clan from memory cache or from table if not present.
-     * @param transaction the tx
-     * @param id ID of clan requested
-     * @return the clan you asked for idiot
-     * @throws IllegalStateException if the clan isn't present in table
+     * Gets a clan by it's identifier
+     *
+     * This fetches from the cache.
+     * @param transaction the transaction
+     * @param id the clan's id
+     * @return the clan if present, empty if not.
      */
-    public Clan getClan(Transaction transaction, int id) {
-        return clans.get(id, i -> {
+    public Optional<Clan> getClanByID(Transaction transaction, int id) {
+        return getClan(transaction, transaction.getProperty(DSLContext.class).fetchOne(CLANS_CLAN_INFO,CLANS_CLAN_INFO.CLAN_ID.eq(id)));
+    }
 
-            var jooq = transaction.getProperty(DSLContext.class);
+    /**
+     * Gets a clan by the name of the clan.
+     *
+     * IMPORTANT: This does not fetch from the cache.
```

Comment:
As much as possible, implementation caching is not something the API user should have to deal with. It's okay if there is no caching in the implementation, but unless the API user doesn't need to know about it, don't note it.

Reply:
> but unless the API user doesn't need to know about it, don't note it. 
 
 Did you mean unless they need to know about it? I'd say it's important if they know about the implementation and whether it uses caching or not, but i can remove it from documentation if you want

Reply:
I meant "unless the API user needs to know about it, don't note it."

I think it would make more sense for the documentation to omit implementation details like caching, which should affect performance only.